### DISCUSSION TITLE
k8s native rules operator based model

### DISCUSSION THREAD

**Discussion by robpearce-flux** [jertel/elastalert2#1234] [Q&A] [answered]
Looking at how the helm chart of elastalert2 works, it seems to want to favour having a single secret with rules defined, and then you need to specify the names of the rules in the secret you want to load from.

It would be a lot nicer if it were possible to have a CRD for an alert rule, and then an operator that collects those types together so they can be made dynamically on the fly, and by multiple users of the cluster. This would allow end-users to define their own rules and alerts as part of their application deployment helm charts, similar to the prometheus operator and how that works for metrics based alerting.

Has anyone made any headway already on creating something like that for elastalert2 ? It'd be *super* handy, imho.

**Comment by jertel** [comment, marked as answer]
The chart supports other methods for loading rules but I'm unaware of any efforts to do so via CRD. Here are some discussions that may give you some ideas.

 #507

#528

 #1134

**Reply by robpearce-flux** [reply]
hmm, i'll take a look. Our cluster is entirely stateless so i wouldn't want to be writing out to volumes and the app teams don't have permissions to alter the platform namespaces, so i was hoping i'd be able to let them deploy a crd into their own namespace that's kept with their own app's codebase. Imho it'd be a much more k8s native way of operating and make the whole thing feel much more seamless.

I'll see if i can do something with the options you highlighted though, thanks.

**Reply by jertel** [reply]
I see where you're coming from and don't disagree with the CRD technique. If you're interested in creating the templates and testing it I'd be open to including it in the project for the next release.

**Reply by jertel** [reply]
Also, [@mehrdad-khojastefar](https://github.com/mehrdad-khojastefar) just today posted PR #1236 which generates the rules config during chart installation. Since they also need a more flexible method of deploying the rules for a k8s deployment perhaps you both would be interested in collaborating on solution.
